pkg/sql/schemachanger/scexec/scmutationexec/scmutationexec.go

```diff
@@ -262,19 +262,26 @@ func (m *visitor) AddTypeBackRef(ctx context.Context, op scop.AddTypeBackRef) er
 func (m *visitor) RemoveRelationDependedOnBy(
 	ctx context.Context, op scop.RemoveRelationDependedOnBy,
 ) error {
-	// Remove the descriptors namespaces as the last stage
+	// Clean up dependencies for the relationship.
 	tbl, err := m.checkOutTable(ctx, op.TableID)
 	if err != nil {
 		return err
 	}
-	for depIdx, dependedOnBy := range tbl.DependedOnBy {
-		if dependedOnBy.ID == op.DependedOnBy {
+	depDesc, err := m.checkOutTable(ctx, op.DependedOnBy)
+	if err != nil {
+		return err
+	}
+	for depIdx, dependsOnBy := range tbl.DependedOnBy {
+		if dependsOnBy.ID == op.DependedOnBy {
 			tbl.DependedOnBy = append(tbl.DependedOnBy[:depIdx], tbl.DependedOnBy[depIdx+1:]...)
 			break
 		}
 	}
-	if len(tbl.DependedOnBy) == 0 {
-		tbl.DependedOnBy = nil
```

Comment:
I should have left a comment, but I put this nil-setting code here for a reason. See my comment about `[]` entries in the end-to-end test expected output.

Reply:
Let me revert that and comment on this. It seemed unnecessary, but now it all makes sense.